pkg/userdata/coreos/userdata.go

```diff
@@ -21,15 +21,15 @@ type config struct {
 }
 
 func getConfig(r runtime.RawExtension) (*config, error) {
-	p := config{}
+	p := new(config)
 	if len(r.Raw) == 0 {
-		return &p, nil
+		return p, nil
 	}
 
-	if err := json.Unmarshal(r.Raw, &p); err != nil {
+	if err := json.Unmarshal(r.Raw, p); err != nil {
```

Comment:
This also decreases readability with the fact that you pass a pointer. The old version displays this very clearly... :confused: